overlord/devicestate/devicemgr.go

```diff
@@ -1360,6 +1372,44 @@ func (m *DeviceManager) Registered() <-chan struct{} {
 	return m.reg
 }
 
+type UnregisterOptions struct {
+	NoRegistrationUntilReboot bool
+}
+
+// Unregister unregisters the device forgetting its serial
+// plus the additional behavior described by the UnregisterOptions
+func (m *DeviceManager) Unregister(opts *UnregisterOptions) error {
```

Comment:
This is called from contexts where the caller already has the state lock?

Reply:
yes